shared/modules/contract-utils.ts

```diff
@@ -1,21 +1,29 @@
-type contract = {
+import pify from 'pify';
+
+export type Contract = {
   contractCode: string | null;
   isContractAddress: boolean;
 };
 
+// Note(@dbrans): This is a simplified version of the 'EthQuery' interface specific to this file.
+type EthQueryWithGetCode = {
+  getCode: (
+    address: string,
+    cb: (err: Error, contractCode: string) => void,
+  ) => void;
+};
+
 export const readAddressAsContract = async (
-  ethQuery: {
-    getCode: (address: string) => string;
-  },
+  ethQuery: EthQueryWithGetCode,
   address: string,
-): Promise<contract> => {
-  let contractCode;
+): Promise<Contract> => {
+  let contractCode: string | null;
   try {
-    contractCode = await ethQuery.getCode(address);
-  } catch (e) {
+    contractCode = await pify(ethQuery.getCode.bind(ethQuery))(address);
+  } catch (err) {
+    // TODO(@dbrans): Dangerous to swallow errors here.
```

Comment:
@dbrans if there isn't a issue for this already its good to create one so we don't end up losing these.

Reply:
Thanks @brad-decker. Created https://github.com/orgs/MetaMask/projects/47/views/1?pane=issue&itemId=43336279. Will refine.